app/src/main/java/com/android/feedme/model/data/CommentRepository.kt

```diff
@@ -35,13 +38,40 @@ class CommentRepository(private val db: FirebaseFirestore) {
    * @param onSuccess Callback invoked on successful addition of the comment.
    * @param onFailure Callback invoked on failure to add the comment, with an exception.
    */
-  fun addComment(comment: Comment, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+  fun addComment(
+      comment: Comment,
+      uri: Uri?,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
     val newDocRef = db.collection("comments").document()
     comment.commentId = newDocRef.id // Assign the generated ID to the comment
-    newDocRef
-        .set(comment)
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception -> onFailure(exception) }
+    if (uri != null) {
+      val storageRef =
+          FirebaseStorage.getInstance().reference.child("comments/${comment.commentId}")
```

Comment:
Locating the 'comments/' path within the code is challenging. To enhance maintainability, it would be better to store this path in a variable at the beginning of the document, just like in `ProfileRepository.kt.`